src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/PropertyAccessorSymbolReferenceFinder.cs

```diff
@@ -1,47 +1,94 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
-using System.Collections.Generic;
 using System.Collections.Immutable;
 using System.Threading;
 using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.LanguageServices;
 using Microsoft.CodeAnalysis.Shared.Extensions;
-using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.FindSymbols.Finders
 {
     internal class PropertyAccessorSymbolReferenceFinder : AbstractMethodOrPropertyOrEventSymbolReferenceFinder<IMethodSymbol>
     {
         protected override bool CanFind(IMethodSymbol symbol)
-        {
-            return symbol.MethodKind.IsPropertyAccessor();
-        }
+            => symbol.MethodKind.IsPropertyAccessor();
 
         protected override async Task<ImmutableArray<SymbolAndProjectId>> DetermineCascadedSymbolsAsync(
             SymbolAndProjectId<IMethodSymbol> symbolAndProjectId,
             Solution solution,
             IImmutableSet<Project> projects,
+            FindReferencesSearchOptions options,
             CancellationToken cancellationToken)
         {
             var result = await base.DetermineCascadedSymbolsAsync(
-                symbolAndProjectId, solution, projects, cancellationToken).ConfigureAwait(false);
+                symbolAndProjectId, solution, projects, options, cancellationToken).ConfigureAwait(false);
 
-            var symbol = symbolAndProjectId.Symbol;
-            if (symbol.AssociatedSymbol != null)
+            // If we've been asked to search for specific accessors, then do not cascade.
+            // We don't want to produce results for the associated property.
+            if (!options.AssociatePropertyReferencesWithSpecificAccessor)
             {
-                result = result.Add(symbolAndProjectId.WithSymbol(symbol.AssociatedSymbol));
+                var symbol = symbolAndProjectId.Symbol;
+                if (symbol.AssociatedSymbol != null)
+                {
+                    result = result.Add(symbolAndProjectId.WithSymbol(symbol.AssociatedSymbol));
+                }
             }
 
             return result;
         }
 
-        protected override Task<ImmutableArray<Document>> DetermineDocumentsToSearchAsync(IMethodSymbol symbol, Project project, IImmutableSet<Document> documents, CancellationToken cancellationToken)
+        protected override async Task<ImmutableArray<Document>> DetermineDocumentsToSearchAsync(
+            IMethodSymbol symbol, Project project, IImmutableSet<Document> documents, 
+            FindReferencesSearchOptions options, CancellationToken cancellationToken)
         {
-            return FindDocumentsAsync(project, documents, cancellationToken, symbol.Name);
+            var result = await FindDocumentsAsync(
+                project, documents, cancellationToken, symbol.Name).ConfigureAwait(false);
```

Comment:
you have actually found a bug here.  we should be concatenating these arrays.  I will doc, but the first search is for looking for actual usage of things like get_Foo in the code.  This can happen in cross language scenarios when C# references a VB parameterized property.    The second part is looking for accessor references through the property.  Both can be found, and we should report all of them.